src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,398 @@
+        # command if we know the hostname, so create the IP - hostname mapping.
+        ip_to_host = self.create_ip_to_host()
+        for member in members:
+            ip_addr = member["addr"].split(":")[0]
+            if rank_0_ip != ip_addr:
+                self.network_down_host = ip_to_host[ip_addr]
+                break
+        self.log.info("network_down_host = %s", self.network_down_host)
+
+        # 2. Create a pool across two ranks on the same node; 0 and rank_r. We have to
+        # provide the size because we're using --ranks.
+        self.add_pool(namespace="/run/pool_size_value/*", create=False)
+        self.pool.target_list.update([0, rank_r])
+        self.pool.create()
+
+        # 3. Create a container without redundancy factor.
+        self.container = []
+        self.container.append(
+            self.get_container(pool=self.pool, namespace="/run/container_wo_rf/*"))
+
+        # 4. Take down the interface where the pool isn't created.
+        errors = []
+        self.interface = self.params.get(
+            "fabric_iface", "/run/server_config/servers/0/*")
+
+        # wolf
+        if self.test_env == "ci":
+            self.update_network_interface(
+                interface=self.interface, state="down", host=self.network_down_host,
+                errors=errors)
+        else:
+            # Aurora. Manually run the command.
+            command = "sudo ip link set {} {}".format(self.interface, "down")
+            self.log.debug("## Call %s on %s", command, self.network_down_host)
+            time.sleep(20)
+
+        # 5. Run IOR with oclass SX.
+        ior_results = {}
+        job_num = 1
+        ior_namespace = "/run/ior_wo_rf/*"
+        self.run_ior_report_error(
+            job_num=job_num, results=ior_results, file_name="test_file_1",
+            pool=self.pool, container=self.container[0], namespace=ior_namespace)
+
+        # 6. Verify that IOR worked.
+        self.verify_ior_worked(ior_results=ior_results, job_num=job_num, errors=errors)
+
+        # 7. Verify that the container Health is HEALTHY.
+        if not self.container[0].verify_health(expected_health="HEALTHY"):
+            errors.append(
+                "Container health isn't HEALTHY after taking ib0 down!")
+
+        # 8. Create a new container on the pool and run IOR.
+        self.container.append(
+            self.get_container(pool=self.pool, namespace="/run/container_wo_rf/*"))
+
+        # Run IOR and verify that it works.
+        job_num = 2
+        self.run_ior_report_error(
+            job_num=job_num, results=ior_results, file_name="test_file_2",
+            pool=self.pool, container=self.container[1], namespace=ior_namespace)
+        self.verify_ior_worked(ior_results=ior_results, job_num=job_num, errors=errors)
+
+        # 9. Bring up the network interface.
+        if self.test_env == "ci":
+            # wolf
+            self.update_network_interface(
```

Comment:
(pylint-no-member) Instance of 'NetworkFailureTest' has no 'update_network_interface' member